### PR TITLE
Add Neural-search to 3.0.0 manifest

### PR DESCRIPTION
**Pull request by martin-gaievski**
### Description
Add neural-search plugin to 3.0.0 manifest

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaiksaya** [comment]
Looks like the CI passed. I assume neural does not need ml-commons as a dependency during the build since this manifest lacks it?

**Comment by martin-gaievski** [comment]
> Looks like the CI passed. I assume neural does not need ml-commons as a dependency during the build since this manifest lacks it?

We use ml-commons jars from public repo, as to my knowledge ml-commons is doing GHA to publish artifacts to maven, it's independent of the distribution build.

Currently ml-commons is unblocked, I can add it with this PR if that's ok

**Comment by gaiksaya** [comment]
> > Looks like the CI passed. I assume neural does not need ml-commons as a dependency during the build since this manifest lacks it?
> 
> We use ml-commons jars from public repo, as to my knowledge ml-commons is doing GHA to publish artifacts to maven, it's independent of the distribution build.
> 
> Currently ml-commons is unblocked, I can add it with this PR if that's ok

Yes please do. Maven only has snapshots version of ml-commons. However while building distribution, the snapshot is set to false. I believe it would fail then since it is unable to find it in  maven local.

**Comment by martin-gaievski** [comment]
Added ml-commons as requested, fyi @b4sjoo

File: manifests/3.0.0/opensearch-3.0.0.yml [hunk 1]
```diff
@@ -23,6 +23,15 @@ components:
...
       - gradle:properties:version
+  - name: ml-commons
+    repository: https://github.com/opensearch-project/ml-commons.git
+    ref: main
+    platforms:
+      - linux
+      - windows
+    checks:
+      - gradle:properties:version
+      - gradle:dependencies:opensearch.version: opensearch-ml-plugin
```
Comment (gaiksaya) [inline manifests/3.0.0/opensearch-3.0.0.yml:34]: Not sure if JS is one of the ml-commons's dependency? Care to make it below job-scheduler? Just to be safe?
Thanks!

Comment (martin-gaievski) [reply manifests/3.0.0/opensearch-3.0.0.yml:34]: order ml-commons - js was in the manifest before it got removed last time, basically I restored manifest to the form it was before that change 5 days ago - https://github.com/opensearch-project/opensearch-build/commit/7cc32169662445fbd1d177153c2b459573cd7cb7

**Review by gaiksaya — APPROVED** [review]

**Pull request merged**